src/broker/Manager.cc

```diff
@@ -1221,10 +1216,16 @@ void Manager::Process()
 	if ( had_input )
 		{
 		if ( run_state::network_time == 0 )
+			{
 			// If we're getting Broker messages, but still haven't initialized
 			// run_state::network_time, may as well do so now because otherwise the
 			// broker/cluster logs will end up using timestamp 0.
-			run_state::detail::update_network_time(util::current_time());
+			//
+			// Do not do this when allow_network_time_forard is set to F
+			// with the assumption that this is unwanted behavior.
+			if ( get_option("allow_network_time_forward")->AsBool() )
+				run_state::detail::update_network_time(util::current_time());
```

Comment:
Possibly, yes. I'm going to leave this as is for now: It may actually not be needed anymore as we'll have done the time update in the main-loop if there wasn't actually a packet on a worker (and would've done it on a non-worker) before the first `event_mgr.Drain()`.

I might just remove it in a follow-up directly.